sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala

```diff
@@ -781,3 +783,64 @@ case class SchemaOfJson(
 
   override def prettyName: String = "schema_of_json"
 }
+
+/**
+ * A function which returns all the keys of outer JSON object.
+ */
+@ExpressionDescription(
+  usage = "_FUNC_(json_object) - returns all the keys of outer JSON object.",
+  arguments = """
+    Arguments:
+      * json_object - A JSON object is required as argument. `Null` is returned, if an invalid JSON
+          string is given. `Analysis Exception` is thrown, if null string or JSON array is given.
+  """,
+  examples = """
+    Examples:
+      > Select _FUNC_('{}');
+        []
+      > Select _FUNC_('{"key": "value"}');
+        ["key"]
+      > Select _FUNC_('{"f1":"abc","f2":{"f3":"a", "f4":"b"}}');
+        ["f1","f2"]
+  """,
+  since = "3.1.0")
+case class JsonObjectKeys(child: Expression) extends UnaryExpression with CodegenFallback {
+  override def dataType: DataType = ArrayType(StringType)
+  override def nullable: Boolean = true
+  override def prettyName: String = "json_object_keys"
+
+  private lazy val json = child.eval().asInstanceOf[UTF8String]
+
+  override def eval(input: InternalRow): Any = {
+    try {
+      Utils.tryWithResource(CreateJacksonParser.utf8String(SharedFactory.jsonFactory, json)) {
+        parser => getJsonKeys(parser, input)
+      }
+    } catch {
+      case _: JsonProcessingException => null
+    }
+  }
+
+  private def getJsonKeys(parser: JsonParser, input: InternalRow): Any = {
+    var arrayBufferOfKeys = ArrayBuffer.empty[UTF8String]
+    // this handles `NULL` case
+    if (parser.nextToken() == null) {
+      throw new AnalysisException(s"$prettyName expect a JSON object but nothing is provided.")
```

Comment:
Is the exception thrown at analysis phase? I think `AnalysisException` should be replaced by `RuntimeException`.